agent/bench-scripts/driver/linpack

```diff
@@ -0,0 +1,277 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: t; sh-basic-offset: 8; sh-indentation: 8; tab-width: 8 -*-
+#
+# "linpack" driver script
+#
+# Responsible for transforming pbench-linpack command line inputs into the
+# the linpack invocation requirements.
+#
+# The linpack binary expects an input file with 7 lines in it:
+#
+#     Line 1: Header (disgarded)
+#     Line 2: Sub-Header (emitted as-is to stdout)
+#     Line 3: Positive integer number of tests to run
+#     line 4: Number of equations to solve (problem size)
+#     line 5: Leading dimension of array
+#     line 6: Number of trials to run
+#     line 7: Data alignment value (in Kbytes)
+#
+# For lines 4-7, if there are N tests to run, each line should have N values
+# describing the parameters for those tests. For example:
+#
+#     Line 1: Header
+#     Line 2: Sub-Header
+#     Line 3: 3
+#     line 4: 10000 20000 30000
+#     line 5: 10016 20016 30016
+#     line 6:     2     2     2
+#     line 7:     4     4     4
+#
+# This script provides optional input parameters for lines 1, 2, 4 - 7, and
+# then calculates the number of tests to be run from those inputs to fill in
+# line 3 of the LINPACK input file.
+#
+# Additionally, it offers input parameters to control the use of OpenMP during
+# the LINPACK run, along with the OMP_NUM_THREADS and KMP_AFFINITY environment
+# variables used by OpenMP, and if any NUMA controls should be applied (via
+# `numactl`).  These parameters are provided in the default "Sub-Header".
+#
+# The script generates three files:
+#
+#    ${output_dir}/linpack.input  -  Constructed seven line LINPACK input file
+#    ${output_dir}/linpack.meta   -  All input parameters provided as key=val
+#    ${output_dir}/linpack.out    -  The output generated by the LINPACK binary
+#
+
+script_name="$(basename ${0})"
+
+linpack_input_file_name="linpack.input"
+linpack_metadata_file_name="linpack.meta"
+linpack_output_file_name="linpack.out"
+
+linpack_binary=""
+
+# Input file defaults, only lines 4-7 (headers defined below).
+header=""
+subheader=""
+
+# Line 4 default
+problem_sizes=20000
+# Line 5 default
+leading_dimensions=20016
+# Line 6 default
+run_trials=1
+# Line 7 default
+alignment_values=4
+
+# Environment input defaults
+use_omp="y"
+threads_arg=""
+kmp_affinity_args="nowarnings,compact,1,0,granularity=fine"
+numactl_args=""
+
+# Output directory required.
+output_dir=""
+
+function usage() {
+	printf -- "TBD -- how to use the pbench linpack driver ...\n"
```

Comment:
Are you planning to change this before merging?

Reply:
I wasn't because it is not part of the pbench-agent command line interface, and was taking the easy way out since this is just a refactor of the previous incarnation of this script, `linpack-prepare-input-file`.

I'll take a stab at a usage message.